cpp/src/arrow/util/value_parsing_test.cc

```diff
@@ -838,12 +838,25 @@ TEST(TimestampParser, StrptimeZoneOffset) {
   std::string format = "%Y-%d-%m %H:%M:%S%z";
   auto parser = TimestampParser::MakeStrptime(format);
 
+  std::vector<std::string> values = {
+    "2018-01-01 00:00:00+0000",
+    "2018-01-01 00:00:00+0100",
+#if defined(__GLIBC__) && defined(__GLIBC_MINOR__)
+// glibc < 2.28 doesn't support "-0117" timezone offset.
+// See also: https://github.com/apache/arrow/issues/43808
+#  if (__GLIBC__ >= 2) && (__GLIBC_MINOR__ >= 28)
```

Comment:
Ah, sorry. This condition was wrong..

```suggestion
#  if ((__GLIBC__ == 2) && (__GLIBC_MINOR__ >= 28)) || (__GLIBC__ >= 3)
```

Reply:
Given that there's no glibc 3 on the horizon, I wonder if this is worth it, but please feel free to commit this if you want to future-proof things longterm. I'd be willing to bet a small amount that arrow will require `glibc >=2.28` before glibc 3.0 gets released. ;-)